### DISCUSSION TITLE
FQDN, Alias and Allowed Origin ambiguity

### DISCUSSION THREAD

**Discussion by fhopley** [llamafilm/tesla-http-proxy-addon#33] [General]
I think there is some ambiguity in the instructions which maybe somebody could clarify for me.
In the instructions it states that you need:
* a fully qualified domain name (FQDN) - and the example name is given as home.example.com
* an alias pointing at the same IP address - and the example name is given as tesla.example.com

In the Allowed Origin it states you should use the FQDN of your Home Assistant server, but it gives the example as https://tesla.example.com

This seems ambiguous to me, as the FQDN example name is home.example.com NOT tesla.example.com 
I've followed the instructions through right to the end both ways, but neither worked and I'd like to focus troubleshooting on one or other path.  Would be great to know what was intended.

The way that I generated the two domains, both pointing at the same IP address was to create two domains on duckdns.org (both pointing to my HA server external IP address - something like aaa.duckdns.org & aaatesla.duckdns.org).  I put both of these domains into the duckdns add-on, but only put one into the NGINX Home Assistant SSL Proxy as I didn't see how to add a second domain there.  Writing this out it feels likely that this isn't the intended use, and that I will need a real top level domain (which I'm happy to get) and then maybe only put the top level domain (example.com) into both the duckdns and nginx add-on?

Any help is much appreciated, the thought of backing out all the functionality I've built on top of the Tesla Custom integration out of HA is heartbreaking.

**Comment by llamafilm** [comment]
This addon needs to be reachable at https://tesla.example.com on port 443. 

The other domain home.example.com actually doesn’t matter. I just mentioned that because I thought it will simplify this addon config, if you already have a working SSL setup to access Home Assistant. Now I’m realizing some folks have other more exotic ways of accessing HA remotely. So I’m open to ideas on improving the docs.

**Reply by ps121988** [reply]
Hello 

I have nginx integration using a different port from 443.  I use it to access HA from internet using duckdns as domain.
Could you please explain how to add another FQDN ? 
thank you

**Comment by fhopley** [comment]
Thanks for replying.  In that case I'll try set everything up with a single domain name (in duckdns add-on, same name in NGINX and same name again in Tesla http proxy).  In my case in the past I had no SSL setup to access HA and was trying to set it up specifically to get the Tesla custom integration working again.

**Comment by fhopley** [comment]
Trying to progress this, I've got to the point where my public key should be accessible at https://tesla.example.com/.well-known/appspecific/com.tesla.3p.public-key.pem

My domain (https://tesla.example.com) is accessible and shows the HA login screen.  The certificate is valid.
The Tesla http proxy add-on first run looked successful from the logs (extract below).
I restarted NGINX add-on to pick up the new config (with active: true).
However, trying to access the public key location I get "404: Not Found"

I've had a look at /share/nginx_proxy/nginx_tesla.conf which includes all the lines of config specified in the logs of the first run.  There are also two files at /share/tesla (com.tesla.3p.public-key.pem & selfsigned.pem) - and the first of these files includes the public key which is supposed to be accessible.

The logs of the nginx add-on include this error:
nginx: [warn] conflicting server name "tesla.example.com" on 0.0.0.0:443, ignored

So not sure why I have a conflict as it doesn't say what it conflicts with. My application access with Tesla has been granted, and I have a client ID & client secret, but obviously not much point progressing until I get nginx to make the public key accessible.

Any help would be much appreciated.  Reminder again that I'm doing this with just a single domain (for this express purpose) in case that could be related. 

`[17:50:46] INFO: Generating keypair
-----BEGIN PUBLIC KEY-----
MFkwE....
o6OYE....
-----END PUBLIC KEY-----
[17:50:46] NOTICE: Request application access with Tesla, then fill in credentials and restart addon.`

**Comment by fhopley** [comment]
It would appear that the issue I was facing was that the domain specified in the NGINX add-on primary configuration CANNOT be exactly the same as  the domain used in the Tesla http proxy (which modifies the NGINX config with details about tesla.example.com).  This is the conflict of the server name referred to in the nginx add-on logs.  To work around this I changed the domain in the NGINX add-on, and then the conflict was resolved and the public key is now accessible.

After resolving this the remainder of the process worked until I got to the point of having to configure the Tesla Custom Integration.  To my amazement I found that Tesla must have (temporarily?) rolled back access to the owner API as the custom integration was working normally again.  So it seems I will likely have to wait until Tesla removes access to the owner API again before being able to progress.  I noticed originally when I lost access to the owner API that the Tesla app showed the 'Charge on Solar' functionality, but that functionality has now disappeared.

My concern about how I have addressed the nginx add-on issue is that my HA instance is no longer fully available on tesla.example.com - the site shows a HA landing screen with an error 'Unable to fetch auth providers. https://tesla.example.com/?auth_callback=1'  I'm not sure if this will break things in the final stages.

Is there a way for me to test the final pieces of this before I lose access to the owner API?

**Reply by IanK6449** [reply]
Hi FHopley,

I think I am in the same situation that you are / were facing.
I was already using DuckDNS to access my HASS instance remotely but was using a different external port (i.e. not 443) to access it. This all worked fine.
I tried to follow the instructions for the Tesla HTTP Proxy addon but was running into the same issues.
You said you got it to work by changing the domain name in the NGINX addon but you didn't mention what to? 
So, do you still only have a single domain defined in DuckDNS or are you now using two and you've used the second one (tesla.example.com) in the NGINX addon?
